test/e2e/task/start_test.go

```diff
@@ -83,13 +84,83 @@ Waiting for logs to be available...
 		assert.Assert(t, is.Regexp(expected, res.Stdout()))
 	})
 
+	t.Run("Start TaskRun using tkn task start command with --use-param-defaults and all the params having default", func(t *testing.T) {
+		res := tkn.Run("task", "start", "read-task",
+			"-i=source="+tePipelineGitResourceName,
+			"--use-param-defaults",
+			"-p=FILENAME=README.md",
+			"--showlog",
+			"true")
+
+		res.Assert(t, icmd.Expected{
+			ExitCode: 0,
+			Err:      icmd.None,
+		})
+		assert.Assert(t, is.Regexp("TaskRun started:.*", res.Stdout()))
+	})
+
+	t.Run("Start TaskRun using tkn task start command with --use-param-defaults and some of the params not having default", func(t *testing.T) {
+		tkn.RunInteractiveTests(t, &e2e.Prompt{
+			CmdArgs: []string{"task", "start", "read-task",
+				"-i=source=" + tePipelineGitResourceName,
+				"--use-param-defaults",
+				"--showlog",
+				"true"},
+			Procedure: func(c *expect.Console) error {
+
+				if _, err := c.ExpectString("Value for param `FILENAME` of type `string`?"); err != nil {
+					return err
+				}
+
+				if _, err := c.SendLine("README.md"); err != nil {
+					return err
+				}
+
+				if _, err := c.Send(string(terminal.KeyEnter)); err != nil {
+					return err
+				}
+
+				if _, err := c.ExpectEOF(); err != nil {
+					return err
+				}
+
+				c.Close()
+				return nil
+			}})
+	})
+
 	t.Run("Get list of TaskRuns from namespace  "+namespace, func(t *testing.T) {
+
+		taskRun1GeneratedName := e2e.GetTaskRunListWithName(c, "read-task").Items[0].Name
+		taskRun2GeneratedName := e2e.GetTaskRunListWithName(c, "read-task").Items[1].Name
+		taskRun3GeneratedName := e2e.GetTaskRunListWithName(c, "read-task").Items[2].Name
+
+		if err := e2e.WaitForTaskRunState(c, taskRun1GeneratedName, e2e.TaskRunSucceed(taskRun1GeneratedName), "TaskRunSucceed"); err != nil {
+			t.Errorf("Error waiting for TaskRun to Succeed: %s", err)
+		}
+
+		if err := e2e.WaitForTaskRunState(c, taskRun2GeneratedName, e2e.TaskRunSucceed(taskRun2GeneratedName), "TaskRunSucceed"); err != nil {
+			t.Errorf("Error waiting for TaskRun to Succeed: %s", err)
+		}
+
+		if err := e2e.WaitForTaskRunState(c, taskRun2GeneratedName, e2e.TaskRunSucceed(taskRun3GeneratedName), "TaskRunSucceed"); err != nil {
```

Comment:
@danielhelfand previously I tried to show  all the params in the describe command for taskrun (whether default value for param is provided or not),  but now I have modified the code to show only `params` which do not have `default` value defined aka params which we provide in the interactive prompt. 

Done this change based on the suggestion from 
@piyush-garg 
>  I think we don't need to add params section in taskrun if the --use-param-defaults is used, and the controller will do the work.

and from @vdemeester 
> So indeed, we don't really need to "add" the default values when building the Run object stuck_out_tongue_closed_eyes

@piyush-garg @vdemeester Please correct me if I understood anything different :sweat_smile:

Reply:
I understand the reasoning for not specifying default values on the TaskRun object, but I guess it has an unintended effect on `tkn taskrun describe` since the command is expecting the TaskRun to have the param values specified under the TaskRun spec. If param values, whether default or not, are not on TaskRun spec, then the issue is that `tkn tr desc` doesn't show param values specified at run time.

I think your implementation is fine, but we should not introduce something that will break other commands. We may need to look into refactoring taskrun/pipelinerun describe commands to expect that runtime values of params can appear elsewhere. Or pipelines should potentially also apply the param values server side on the spec of the runtime objects.

Reply:
> I think your implementation is fine, but we should not introduce something that will break other commands. We may need to look into refactoring taskrun/pipelinerun describe commands to expect that runtime values of params can appear elsewhere. Or pipelines should potentially also apply the param values server side on the spec of the runtime objects.

So, I am fine *breaking* the `tkn taskrun desc` command over this, mainly because if the user creates the `TaskRun` himself (relying on default parameters too), it will show the same way as this PR makes it — so although it changes *how* `TaskRun` are sent, it's consistent with what happens when `tkn` did not start the `TaskRun`. 

We need an issue to track work on "refining" the `desc` command but I don't think we should block this PR for this.